pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BinaryOperatorTransformFunction.java

```diff
@@ -71,218 +113,331 @@ public TransformResultMetadata getResultMetadata() {
+    String[] rightStringValues = _rightTransformFunction.transformToStringValuesSV(projectionBlock);
+    for (int i = 0; i < length; i++) {
+      try {
+        _results[i] =
+            getIntResult(BigDecimal.valueOf(leftValues[i]).compareTo(new BigDecimal(rightStringValues[i])));
+      } catch (NumberFormatException e) {
+        _results[i] = 0;
+      }
+    }
+  }
+
+  private void fillIntResultArray(ProjectionBlock projectionBlock, double[] leftValues, int length) {
+    int[] rightIntValues = _rightTransformFunction.transformToIntValuesSV(projectionBlock);
+    for (int i = 0; i < length; i++) {
+      _results[i] = getIntResult(Double.compare(leftValues[i], rightIntValues[i]));
+    }
+  }
+
+  private void fillLongResultArray(ProjectionBlock projectionBlock, double[] leftValues, int length) {
+    long[] rightValues = _rightTransformFunction.transformToLongValuesSV(projectionBlock);
+    for (int i = 0; i < length; i++) {
+      _results[i] = getIntResult(BigDecimal.valueOf(leftValues[i]).compareTo(BigDecimal.valueOf(rightValues[i])));
+    }
+  }
+
+  private void fillFloatResultArray(ProjectionBlock projectionBlock, double[] leftValues, int length) {
+    float[] rightFloatValues = _rightTransformFunction.transformToFloatValuesSV(projectionBlock);
+    for (int i = 0; i < length; i++) {
+      _results[i] = getIntResult(Double.compare(leftValues[i], rightFloatValues[i]));
+    }
+  }
+
+  private void fillDoubleResultArray(ProjectionBlock projectionBlock, double[] leftValues, int length) {
+    double[] rightDoubleValues = _rightTransformFunction.transformToDoubleValuesSV(projectionBlock);
+    for (int i = 0; i < length; i++) {
+      _results[i] = getIntResult(Double.compare(leftValues[i], rightDoubleValues[i]));
+    }
+  }
+
+  private void fillStringResultArray(ProjectionBlock projectionBlock, double[] leftValues, int length) {
+    String[] rightStringValues = _rightTransformFunction.transformToStringValuesSV(projectionBlock);
+    for (int i = 0; i < length; i++) {
+      try {
+        _results[i] =
+            getIntResult(BigDecimal.valueOf(leftValues[i]).compareTo(new BigDecimal(rightStringValues[i])));
+      } catch (NumberFormatException e) {
+        _results[i] = 0;
+      }
     }
   }
 
   private int getIntResult(int comparisonResult) {
     return getBinaryFuncResult(comparisonResult) ? 1 : 0;
   }
 
-  protected abstract boolean getBinaryFuncResult(int comparisonResult);
+  private boolean getBinaryFuncResult(int comparisonResult) {
+    switch (_op) {
+      case EQUALS:
+        return comparisonResult == 0;
+      case GREATER_THAN_OR_EQUAL:
+        return comparisonResult >= 0;
+      case GREATER_THAN:
+        return comparisonResult > 0;
+      case LESS_THAN:
+        return comparisonResult < 0;
+      case LESS_THAN_OR_EQUAL:
+        return comparisonResult <= 0;
+      case NOT_EQUAL:
+        return comparisonResult != 0;
+      default:
+        assert false;
```

Comment:
Throw exception?

Reply:
This is impossible unless this class is edited, I don't see the point in handling it any other way. The tests will fail with an assertion error anyway.